ballot/app/test/service.spec.ts

```diff
@@ -1,19 +1,54 @@
-import oasis from '@oasislabs/client';
+const oasis = require('@oasislabs/client');
 
 jest.setTimeout(20000);
 
-describe('Ballot', () => {
-  let service;
+describe('Ballot Test', () => {
+  let ballot;
 
   beforeAll(async () => {
-    service = await oasis.workspace.Ballot.deploy('Secret Ballot', ['Alice', 'Bob'], {
-      header: { confidential: false },
-      gasLimit: '0xe79732',
+    // Deploy your ballot service!
+    ballot = await oasis.workspace.Ballot.deploy('What\'s for dinner', ['beef', 'yogurt'], {
+      header: {confidential: false},
+      gasLimit: '0xf42400',
     });
   });
 
   it('deployed', async () => {
-    expect(service).toBeTruthy();
+    expect(ballot).toBeTruthy();
+  });
+
+  it('test getters', async () => {
+    // Get description
+    const description = await ballot.description();
+    expect(description).toEqual('What\'s for dinner');
+
+    // Get candidates' list
+    const candidates = await ballot.candidates();
+    expect(candidates).toEqual(['beef', 'yogurt']);
+
+    // Check if voting is open (it should be)
+    const votingOpen = await ballot.votingOpen();
+    expect(votingOpen).toEqual(true);
+  });
+
+  it('test voting mechanics', async () => {
+    // Check that you can vote without issue
+    await ballot.vote(0);
+    await ballot.close();
+  });
```

Comment:
can you test voting failure cases or voting twice? these tests are great, but not yet complete